unit-tests/dds/test-option-value.py

```diff
@@ -0,0 +1,116 @@
+        dds.option.from_json( ['1', 0, 'desc'] ).set_value( 20.5 ),
+        RuntimeError, 'not convertible to a signed integer: 20.5' )
+
+with test.closure( 'optional (default) value' ):
+    test.check_false( dds.option.from_json( ['1', 0, 'desc'] ).is_optional() )
+    test.check( dds.option.from_json( ['Asic Temperature',None,-40.0,125.0,0.0,0.0,'Current Asic Temperature (degree celsius)',['optional','read-only']] ).is_optional() )
+    dds.option.from_json( ['4', 'string-value', None, 'desc', ['optional']] )
+    dds.option.from_json( ['5', None, 'default-string-value', 'desc', ['optional']] )  # string type is deduced
+    test.check_throws( lambda:
+        dds.option.from_json( ['a', None, 'desc', ['optional']] ),
+        RuntimeError, 'cannot deduce value type: ["a",null,"desc",["optional"]]' )
+
+
+with test.closure( 'mixed types' ):
+    test.check_equal( dds.option.from_json( ['i', 0, 'desc'] ).value_type(), 'int' )
+    test.check_equal( dds.option.from_json( ['f', 0, 'desc', ['float']] ).value_type(), 'float' )
+    test.check_equal( dds.option.from_json( ['1', 0., 0, 1, 1, 0, 'desc'] ).value_type(), 'float' )
+    test.check_equal( dds.option.from_json( ['2', 0, 0., 1, 1, 0, 'desc'] ).value_type(), 'float' )
+    test.check_equal( dds.option.from_json( ['3', 0, 0, 1., 1, 0, 'desc'] ).value_type(), 'float' )
+    test.check_equal( dds.option.from_json( ['4', 0, 0, 1, 1., 0, 'desc'] ).value_type(), 'float' )
+    test.check_equal( dds.option.from_json( ['5', 0, 0, 1, 1, 0., 'desc'] ).value_type(), 'float' )
+    test.check_equal( dds.option.from_json( ['-1', 0, -1, 4, 1, 0, 'desc'] ).value_type(), 'int' )
+    test.check_equal(           # float, but forced to an int
+        dds.option.from_json( ['3f', 3, 1, 5, 1, 2., '', ['int']] ).value_type(), 'int' )
+    test.check_throws( lambda:  # same, but with an actual float
+        dds.option.from_json( ['3f', 3, 1, 5, 1, 2.2, '', ['int']] ),
+        RuntimeError, 'not convertible to a signed integer: 2.2' )
+
+    dds.option.from_json( ['a', 9223372036854775807, 'desc'] )       # max int64
+    test.check_throws( lambda:
+        dds.option.from_json( ['a', 9223372036854775808, 'desc'] ),  # uint64
+        RuntimeError, 'not convertible to a signed integer: 9223372036854775808' )
+    dds.option.from_json( ['a', -9223372036854775808, 'desc'] )    # min int64
+    # -9223372036854775809 cannot be converted to json
+
+    test.check_equal( dds.option.from_json( ['Brightness',0,-64,64,1,0,'UVC image brightness'] ).value_type(), 'int' )
+
+with test.closure( 'range' ):
+    test.check_throws( lambda:
+        dds.option.from_json( ['3x', 0, 0, -1, 1, 0, ''] ),
+        RuntimeError, 'default value 0 > -1 maximum' )
+    test.check_throws( lambda:
+        dds.option.from_json( ['3x', 0, 2, 3, 1, 0, ''] ),
+        RuntimeError, 'default value 0 < 2 minimum' )
+    test.check_throws( lambda:
+        dds.option.from_json( ['3y', -1, -2, 2, 1, 3, 'bad default'] ),
```

Comment:
What is tested here that isn't tested in the previos check?

Reply:
Not sure :)
Before we do have an invalid max, and default==value, while here it's a valid range and different values.
Needs changing?

Reply:
No need to change